File: java/medium/435_non_overlapping_intervals.java
package java.medium;

import java.util.Arrays;
import java.util.Comparator;

class NonOverlappingIntervals {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, Comparator.comparingInt(a -> a[1]));
        int count = 0;
        int k = Integer.MIN_VALUE;
        for (int[] interval : intervals) {
            int x = interval[0];
            int y = interval[1];

            if (x >= k) {
                k = y;
            } else {
                count++;
            }
        }
        return count;
    }
}

File: go/medium/435_non_overlapping_intervals.go
package medium

import (
	"sort"
)

func EraseOverlapIntervals(intervals [][]int) int {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][1] < intervals[j][1]
	})

	result := 0
	prevEnd := intervals[0][1]
	for i := 1; i < len(intervals); i++ {
		start := intervals[i][0]
		end := intervals[i][1]
		if start >= prevEnd {
			prevEnd = end
		} else {
			result++
		}
	}

	return result
}
